tiledb/sm/query/dense_reader.cc

```diff
@@ -563,7 +563,9 @@ DenseReader::apply_query_condition(
                 it->second,
                 cell_slab.coords_.data());
 
-            for (int32_t i = (int32_t)frag_domains.size() - 1; i >= 0; --i) {
+            for (int32_t i = static_cast<int32_t>(frag_domains.size()) - 1;
```

Comment:
Is there any reason the iterator for what seems to be a vector is a signed integer type?

Reply:
Yes, since we iterate in reverse, the signed type allows the >= 0 condition... unsigned would just wrap around.